node/declaration.ts

```diff
@@ -9,14 +9,19 @@ interface BaseDeclaration {
 
 type DeclarationWithPath = BaseDeclaration & {
   path: string
+  excludePath?: string
 }
 
 type DeclarationWithPattern = BaseDeclaration & {
   pattern: string
+  excludePattern?: string
```

Comment:
I *think* that this will be the name of they in the `toml` as well. from the docs, it looks like we're using camelCase there, but we should verify that before merging this.